src/internal/connector/exchange/api/shared.go

```diff
@@ -56,6 +61,54 @@ func toValues[T any](a any) ([]getIDAndAddtler, error) {
 	return r, nil
 }
 
+func getAddedAndRemovedItemIDs(
+	ctx context.Context,
+	service graph.Servicer,
+	pager itemPager,
+	deltaPager itemPager,
+	oldDelta string,
+	canMakeDeltaQueries bool,
+) ([]string, []string, DeltaUpdate, error) {
+	var (
+		pgr        itemPager
+		resetDelta bool
+	)
+
+	if canMakeDeltaQueries {
+		pgr = deltaPager
+		resetDelta = len(oldDelta) == 0
+	} else {
+		pgr = pager
+		resetDelta = true
+	}
+
+	added, removed, deltaURL, err := getItemsAddedAndRemovedFromContainer(ctx, pgr)
+	// note: happy path, not the error condition
+	if err == nil {
+		return added, removed, DeltaUpdate{deltaURL, resetDelta}, err
+	}
+
+	// If we already tried with a non-delta url, we can return
+	if !canMakeDeltaQueries {
+		return nil, nil, DeltaUpdate{}, err
+	}
+
+	// return error if invalid delta error or oldDelta was empty
+	if !graph.IsErrInvalidDelta(err) || len(oldDelta) == 0 {
+		return nil, nil, DeltaUpdate{}, err
+	}
+
+	// reset deltaPager
+	pgr.reset(ctx)
```

Comment:
The reset behavior isn't well explained, both here and in the interface.  I'm not sure what to expect of the state before and after this call, or what the act of "resetting" the pager is intended to solve.  This not only makes it difficult to grok code flow at this point, but will make future contribution around these expectations difficult to maintain.

Reply:
Add an explanation to the reset interface.